Modules/Scripted/Endoscopy/Endoscopy.py

```diff
@@ -643,7 +661,9 @@ def _hideOnlyInView(viewNodeID, displayableNodes):
                 if not displayNode:
                     continue
                 with slicer.util.NodeModify(displayNode):
-                    displayNode.SetViewNodeIDs(allViewNodeIDs)
+                    if not displayNode.GetNumberOfViewNodeIDs():
+                        # Currently empty / not initialized, so initialize with all views.
+                        displayNode.SetViewNodeIDs(allViewNodeIDs)
```

Comment:
```suggestion
                    displayNode.SetViewNodeIDs(allViewNodeIDs)
```

Since the purpose of the function is to hide in all view expect one, this is not needed.

Reply:
Properly addressing this presents some challenges (see https://github.com/Slicer/Slicer/issues/7434), waiting these are sorted out, the current approach addresses the problem described [here](https://github.com/Slicer/Slicer/pull/7165#issuecomment-1827205890).

I then suggest to keep the behavior of `_hideOnlyInView` consistent with its name.

```suggestion
                    displayNode.SetViewNodeIDs(allViewNodeIDs)
```